michelleezhangg/Personal-Website-Portfolio
# [I-8] Active Link/Navigation Highlighting

## Summary
[[Issue-8]](https://github.com/michelleezhangg/Personal-Website-Portfolio/issues/8#issue-2587681130) Implemented a scroll spy feature for the navbar, highlighting the active link based on the current scroll position.

## Implementation Details
- Added react-scroll library for scroll spy functionality and smooth scrolling
- Updated Narbar and MenuOverlay components to use ScrollLink from react-scroll instead of NavLink

## Testing Instructions
1. Run the application locally
2. Scroll through the page and observe the navbar links
3. Verify that the appropriate link is highlighted as you scroll to different sections
4. Click on navbar links to confirm smooth scrolling to the correct section and that navbar functionality still works.
